File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        num_count = Counter(arr)
        max_value = float("-inf")
        max_value = max((n for n, c in num_count.items() if n == c), default=None)
        return -1 if not max_value else max_value

File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        int res = -1;
        unordered_map<int, int> mp;
        for (int n : arr) {
            mp[n]++;
        }

        for (const auto& [k, v] : mp) {
            if (k == v) {
                res = max(res, k);
            }
        }

        return res;
    }
};